File: robot-factory/typescript/src/Money.ts
export class Money {
  constructor(public readonly amount: number) {}
  static readonly zero = new Money(0);
  plus(other: Money): Money { return new Money(this.amount + other.amount); }
  lessThan(other: Money): boolean { return this.amount < other.amount; }
  equals(other: Money): boolean { return this.amount === other.amount; }
}

File: robot-factory/csharp/src/RobotFactory/Money.cs
namespace RobotFactory;

public readonly record struct Money(decimal Amount) : IComparable<Money>
{
    public static Money Zero => new(0m);
    public static Money operator +(Money a, Money b) => new(a.Amount + b.Amount);
    public static bool operator >(Money a, Money b) => a.Amount > b.Amount;
    public static bool operator <(Money a, Money b) => a.Amount < b.Amount;
    public static bool operator >=(Money a, Money b) => a.Amount >= b.Amount;
    public static bool operator <=(Money a, Money b) => a.Amount <= b.Amount;
    public int CompareTo(Money other) => Amount.CompareTo(other.Amount);
}
